### PR TITLE
Laboratory work #4, Anna Vorontsova - 22FPL2

### PR DESCRIPTION
**Pull request by vorontsann**

### CODE REVIEW

File: lab_4_fill_words_by_ngrams/main.py [hunk 1]
```diff
@@ -28,6 +30,17 @@ def _tokenize(self, text: str) -> tuple[str, ...]:  # type: ignore
         Raises:
             ValueError: In case of inappropriate type input argument or if input argument is empty.
         """
+        if not isinstance(text, str) or not text:
+            raise ValueError('Type input is inappropriate or input argument is empty.')
+
+        preprocessed_text = ""
+        for element in text.lower():
+            if element in "?!.":
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:38]: to the variable, please

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Not bad

File: lab_4_fill_words_by_ngrams/main.py (hunk 1 above)
Comment (vorontsann) [reply lab_4_fill_words_by_ngrams/main.py:38]: fixed

**Review by artyomtugaryov — COMMENTED** [review]
Very good

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**